### PR TITLE
Processed expression parsing from uploaded AnnData files (SCP-4912)

### PR DESCRIPTION
**Pull request by bistline**
#### BACKGROUND & CHANGES
This update adds processed expression extraction to AnnData file uploads.  Since the parsing actually happens during the initial "extract" job, adding `processed_expression` to the command line will tell ingest to parse expression data along with extracting clustering/metadata fragment files for downstream processing.

This update also addresses an bug with `AnnDataFileInfo` objects extracting information from `data_fragments` when the `data_type` value is stored as a string instead of a symbol.

#### MANUAL TESTING
1. Pull branch and boot as normal, including `Delayed::Job` worker
2. Download https://github.com/broadinstitute/scp-ingest-pipeline/blob/development/tests/data/anndata/anndata_test.h5ad and upload to a new study using the "AnnData" UX
    * There is clustering information in the `X_tsne` and `spatial` slots
3. In `development.log`, note that `processed_expression` has been added to the command line for the initial extraction job:
```
:actions:
  :commands:
  - python
  - ingest_pipeline.py
  - "--study-id"
  - 6451311b94ec8f59c754a6d7
  - "--study-file-id"
  - 645135cff6429a66f6d29323
  - "--user-metrics-uuid"
  - 0fc4d8e5-6c4c-422e-94a3-c78801757202
  - ingest_anndata
  - "--ingest-anndata"
  - "--anndata-file"
  - gs://fc-38dbd25a-eea7-4273-849b-732b1e664249/anndata_test.h5ad
  - "--extract"
  - '["cluster", "metadata", "processed_expression"]'
  - "--obsm-keys"
  - '["X_tsne"]'
  :flags: []
  :image_uri: gcr.io/broad-singlecellportal-staging/scp-ingest-pipeline:1.26.0
...
```
4. When the job completes, note that the success email includes information about gene expression data that has been parsed:
```
Total parse time: 1 Minute and 7 Seconds
AnnData file ingest has completed
Gene-level entries created: 4
Clustering and metadata entries are being extracted as specified in your AnnData file.
```
5. Observe that the clustering & metadata jobs launch immediately after
6. (Optional) Once the metadata/clustering jobs complete, confirm you can load the study and search for the gene `Foo`:
Screenshot 2023-05-02 at 12 46 29 PM

### CODE REVIEW

File: test/models/ann_data_ingest_parameters_test.rb [hunk 1]
```diff
@@ -7,12 +7,13 @@ class AnnDataIngestParametersTest < ActiveSupport::TestCase
...
     }
 
+    @file_id = BSON::ObjectId.new
     @ingest_cluster_params = {
       ingest_anndata: false,
       extract: nil,
       obsm_keys: nil,
       ingest_cluster: true,
-      cluster_file: 'gs://test_bucket/_scp_internal/anndata_ingest/h5ad_file_id/h5ad_frag.cluster.X_umap.tsv',
+      cluster_file: "gs://test_bucket/_scp_internal/anndata_ingest/#{@file_id}/h5ad_frag.cluster.X_umap.tsv",
```
[the remark this answers is not in the document]
Comment (bistline) [reply test/models/ann_data_ingest_parameters_test.rb:16]: That actually would have to get done in `scp-ingest-pipeline` - that is where we set the path to the fragment in the bucket.  If I change it here, it won't be able to retrieve the file.

[the remark this answers is not in the document]
Comment (eweitz) [reply test/models/ann_data_ingest_parameters_test.rb:16]: Good to know!  @jlchang, would that take notable effort in Ingest Pipeline?  If not, maybe this is worthwhile soon.

[the remark this answers is not in the document]
Comment (jlchang) [reply test/models/ann_data_ingest_parameters_test.rb:16]: @eweitz That should be do-able! I'll make a note in the raw count ticket to consider this suggestion.

**Review by jlchang — APPROVED** [review]
Functional review - confirmed test case ingests as described.

File: test/models/ann_data_ingest_parameters_test.rb (hunk 1 above)
[the remark this answers is not in the document]
Comment (eweitz) [reply test/models/ann_data_ingest_parameters_test.rb:16]: Much appreciated Jean :)

**Pull request merged**